src/main/scala/org/scalameter/Measurement.scala

```diff
@@ -1,15 +1,19 @@
 package org.scalameter
 
+import org.scalameter.picklers.Pickler
 import scala.collection.Seq
 
 @SerialVersionUID(-2541697615491239986L)
-case class Measurement(value: Double, params: Parameters, data: MeasurementData, units: String) {
-  def complete: Seq[Double] = data.complete
+case class Measurement[T: Pickler](value: T, params: Parameters,
```

Comment:
`Measurer` is now serialized in json based serializers using `Pickler`s, since in theory T can be arbitrary type so Jackson would have a problem (de)serializing it. 
I guess that doing it purely with Jackson would require specifying e.g. type hints or adding some annotations. So that overall looks like more complex solution than just using picklers.